### PR TITLE
Skip `consensus_heights` query in `update_client` when possible

### PR DESCRIPTION
**Pull request by soareschen**
Closes: #1362 

## Description

______

For contributor use:

- [ ] Added a changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer/src/foreign_client.rs [hunk 1]
```diff
@@ -554,6 +555,66 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
         self.build_update_client_with_trusted(target_height, Height::zero())
     }
 
+    fn get_trusted_height_from_client_state(
```
Comment (ancazamfir) [inline relayer/src/foreign_client.rs:558]: Minor suggestion, feel free to ignore. we typically drop `get` from func names in `ibc-rs`. Also it is not always the one from client state. Maybe something like:
```suggestion
    fn solve_trusted_height(
```

**Review by ancazamfir — APPROVED** [review]
lgtm, thanks @soareschen!

File: relayer/src/foreign_client.rs (hunk 1 above)
Comment (adizere) [reply relayer/src/foreign_client.rs]: d6c5f78

**Review by adizere — APPROVED** [review]
Go team!

**Comment by soareschen** [comment]
> Go team!

s/Go/Rust/

**Pull request merged**